### PR TITLE
feat: [CI-5326]: front end changes for hosted VMs and new infra design

### PR DESCRIPTION
**Pull request by d1wilko**
### Summary

Create new flow for Hosted VM infrastructure in CI

Add Cloud infra option

New functionality is hidden behind a feature flag so existing tests will pass

#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 53 Issues 
* Code issue 53 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11252)

**Comment by d1wilko** [comment]
retrigger titlecheck

**Review by tan-nhu — DISMISSED** [review]

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 1]
```diff
@@ -310,6 +315,39 @@ const renderUseFromStageVM = ({
...
+}): JSX.Element => {
+  return (
+    <>
+      {(propagatedStage?.stage?.spec?.platform as Platform)?.os && (
+        <>
+          <Text font={{ variation: FontVariation.FORM_LABEL }} margin={{ bottom: 'xsmall' }}>
+            {getString('pipeline.infraSpecifications.os')}
+          </Text>
+          <Text color="black" margin={{ bottom: 'medium' }}>
+            {(propagatedStage?.stage?.spec?.platform as Platform)?.os}
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:333]: already conditionally checked for value above

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 2]
```diff
@@ -310,6 +315,39 @@ const renderUseFromStageVM = ({
...
+          </Text>
+        </>
+      )}
+      {(propagatedStage?.stage?.spec?.platform as Platform)?.arch && (
+        <>
+          <Text font={{ variation: FontVariation.FORM_LABEL }} margin={{ bottom: 'xsmall' }}>
+            {getString('pipeline.infraSpecifications.architecture')}
+          </Text>
+          <Text color="black" margin={{ bottom: 'medium' }}>
+            {(propagatedStage?.stage?.spec?.platform as Platform)?.arch}
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:343]: already conditionally checked for value above

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 3]
```diff
@@ -597,19 +658,42 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
...
+            values.useFromStage
+          )
+          if (validationPropagatedStage?.stage?.spec?.infrastructure) {
+            set(draft, 'stage.spec.infrastructure', {
+              useFromStage: values.useFromStage
+            })
+            set(draft, 'stage.spec.platform', undefined)
+            set(draft, 'stage.spec.runtime', undefined)
+          }
+          if (validationPropagatedStage?.stage?.spec?.platform && validationPropagatedStage?.stage?.spec?.runtime) {
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:682]: already checked for validationPropagatedStage?.stage?.spec? for runtime

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 4]
```diff
@@ -1080,6 +1170,69 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
...
+        label={
+          <Text
+            tooltipProps={{ dataTooltipId: 'os' }}
+            font={{ variation: FontVariation.FORM_LABEL }}
+            margin={{ bottom: 'xsmall' }}
+          >
+            {getString('pipeline.infraSpecifications.selectOs')}
+          </Text>
+        }
+        name={'os'}
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:1185]: Create a map variable above for 'os' and 'arch'

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 5]
```diff
@@ -1573,629 +1730,699 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
...
+                                        const infraType = val as CIBuildInfrastructureType
+                                        setBuildInfraType(infraType)
+                                        if (infraType === CIBuildInfrastructureType.KubernetesHosted) {
+                                          fetchDelegateDetails()
                                         }
-                                      />
-                                    </Accordion>
+
+                                        // macOs is only supported for VMs - default to linux
+                                        const os =
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:2262]: I think acronyms should be capitalized like we do for VMs and like the label shows 
image

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 6]
```diff
@@ -1121,10 +1122,10 @@ export function StageInputSetFormInternal({
                   )
                 ) : null}
               </>
-            ) : (deploymentStageTemplate.infrastructure as any).type === 'VM' ? (
+            ) : (deploymentStageTemplate.infrastructure as any)?.type === 'VM' ? (
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:1125]: Would recommend using CIBuildInfrastructureType.VM from 75-ci/constants

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 7]
```diff
@@ -987,12 +988,12 @@ export function StageInputSetFormInternal({
...
       )}
 
-      {deploymentStageTemplate.infrastructure && (
+      {(deploymentStageTemplate.infrastructure || (deploymentStageTemplate as any).platform) && (
         <div id={`Stage.${stageIdentifier}.Infrastructure`} className={cx(css.accordionSummary)}>
           <div className={css.inputheader}>{getString('infrastructureText')}</div>
 
           <div className={cx(css.nestedAccordions, css.infraSection)}>
-            {(deploymentStageTemplate.infrastructure as any).type === 'KubernetesDirect' ? (
+            {(deploymentStageTemplate.infrastructure as any)?.type === 'KubernetesDirect' ? (
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:996]: Would recommend using CIBuildInfrastructureType.VM from 75-ci/constants... or the one you created in 70-pipeline/constants

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx (hunk 4 above)
Comment (d1wilko) [reply src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:1185]: I don't think this one should change - every other existing name is a string e.g. harnessImageConnectorRef, poolName etc.

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx (hunk 5 above)
Comment (d1wilko) [reply src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:2262]: I agree with you in principle but the os variable here is actually existing code that I have pulled into a function - not new code - I'm terrified or breaking things lol

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**